From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sleigh-InSPECtor <sleighinspector@outlook.com>
Date: Thu, 23 May 2024 12:59:21 +0930
Subject: [PATCH] 6563: AArch32: (Thumb32) missing DecodeImmShift() defaults to
 32 for type '01' (SRType_LSR ) & '10' (SRType_ASR)

AArch32: (Thumb32) fixed lsrs.w

* added missing DecodeImmShift() defaults to 32 for type '01' (SRType_LSR ) & '10' (SRType_ASR)
---
 Ghidra/Processors/ARM/data/languages/ARMTHUMBinstructions.sinc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Ghidra/Processors/ARM/data/languages/ARMTHUMBinstructions.sinc b/Ghidra/Processors/ARM/data/languages/ARMTHUMBinstructions.sinc
index 57495f8160..da8b2fe253 100644
--- a/Ghidra/Processors/ARM/data/languages/ARMTHUMBinstructions.sinc
+++ b/Ghidra/Processors/ARM/data/languages/ARMTHUMBinstructions.sinc
@@ -509,6 +509,7 @@ ThumbExpandImm12: "#"^imm32           is   immed12_i=1 ; immed12_imm3 & thc0707
 
 @if defined(VERSION_6T2) || defined(VERSION_7)
 
+thLsbImm: "#"^lsb	is imm3_shft=0 & imm2_shft=0 & (thc0405=0b01 | thc0405=0b10) [lsb = 32; ] { tmp:4 = lsb; export tmp; }
 thLsbImm: "#"^lsb	is imm3_shft & imm2_shft [ lsb= (imm3_shft<<2) | imm2_shft; ] { tmp:4 = lsb; export tmp; }
 thMsbImm: "#"^thc0004	is thc0004 { tmp:4 = thc0004; export tmp; }
 thWidthMinus1: "#"^width	is thc0004 [ width = thc0004 + 1; ] { tmp:4 = thc0004; export tmp; }
-- 
2.45.1

